src/Controller/Api/FileController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace App\Controller\Api;
+
+use App\Entity\File;
+use App\Message\DeleteFile;
+use App\Repository\FileRepository;
+
+use Doctrine\ORM\EntityManagerInterface;
+use FOS\RestBundle\Controller\AbstractFOSRestController;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * CRUD API for File Entity.
+ */
+class FileController extends AbstractFOSRestController
+{
+    /**
+     * Delete a Dataset and associated Metadata and Difs.
+     *
+     * @param string                 $id             The id of the File to delete.
+     * @param FileRepository         $fileRepository File entity Repository instance.
+     * @param MessageBusInterface    $messageBus     Symfony messenger bus interface instance.
+     * @param EntityManagerInterface $entityManager  Entity manager interface instance.
+     * @Route("/api/file/{id}", name="pelagos_api_datasets_delete", methods={"DELETE"}, defaults={"_format"="json"})
+     *
+     * @IsGranted("IS_AUTHENTICATED_FULLY")
+     *
+     * @return Response A response object with an empty body and a "no content" status code.
+     */
+    public function deleteFile(string $id, FileRepository $fileRepository, MessageBusInterface $messageBus, EntityManagerInterface $entityManager)
```

Comment:
Instead `string $id, FileRepository $fileRepository` use 'File $file' this way it will check the entity for you, get it if it exists, or throw an error if it is not there.